### PR TITLE
Consume optimization packages using new dependency flow system

### PR DESCRIPTION
**Pull request by hoyosjs**
Now that the packages are getting published to the BAR, consume optimization packages from these channels.

Subscriptions are set as well as the appropriate channels.

### CODE REVIEW

**Comment by adiaaida** [comment]
Just checked for myself, and the packages restore as expected. Looks awesome!

**Comment by hoyosjs** [comment]
@dotnet-bot  test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by hoyosjs** [comment]
Good. Does this conflict with your lock-step PR to prevent breaking builds with the format change?

**Comment by adiaaida** [comment]
When this goes in, the IBC change will momentarily break (since it updates the package), but as the official build hasn't been updated quite yet with the ibcoptimize flag, nothing will break (yet). Let me go test with the new changes put together to confirm that we will be ok.

**Comment by jkoritzinsky** [comment]
So to confirm, we want to merge one of the IBC package update PRs first, then merge your IBC package format change, and then finally enable the ibcoptimize flag.

**Comment by hoyosjs** [comment]
Yes. Also, I'll let #22903 go first, then this one (as it already has a merge conflict), and then we should let @adiaaida format PR (#22799) get in, and then your yaml addition of the ibc flag (#22843)

File: dependencies.props [hunk 1]
```diff
@@ -28,17 +28,13 @@
...
-    <IbcDataCurrentRef>fc12d73e8ea045517fde9b4bd1c89065917a5413</IbcDataCurrentRef>
   </PropertyGroup>
 
   <!-- Tests/infrastructure dependency versions. -->
   <PropertyGroup>
     <PgoDataPackageVersion>99.99.99-master-20190208.1</PgoDataPackageVersion>
     <MicrosoftNETCoreRuntimeCoreCLRPackageVersion>3.0.0-preview-27316-02</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>
     <XunitPackageVersion>2.4.1</XunitPackageVersion>
-    <IbcDataPackageVersion>99.99.99-master-20190208.1</IbcDataPackageVersion>
-    <IbcMergePackageVersion>4.6.0-alpha-00001</IbcMergePackageVersion>
```
Comment (adiaaida) [inline dependencies.props]: We need to add this line back in, I think. The IBCMerge package is separate and doesn't come from darc.

File: dependencies.props [hunk 2]
```diff
@@ -28,17 +28,13 @@
   <PropertyGroup>
     <CoreClrCurrentRef>f31086500dc58cefe7b1275300f6fc07b38c1bcc</CoreClrCurrentRef>
     <BuildToolsCurrentRef>fc12d73e8ea045517fde9b4bd1c89065917a5413</BuildToolsCurrentRef>
-    <PgoDataCurrentRef>fc12d73e8ea045517fde9b4bd1c89065917a5413</PgoDataCurrentRef>
-    <IbcDataCurrentRef>fc12d73e8ea045517fde9b4bd1c89065917a5413</IbcDataCurrentRef>
   </PropertyGroup>
 
   <!-- Tests/infrastructure dependency versions. -->
   <PropertyGroup>
     <PgoDataPackageVersion>99.99.99-master-20190208.1</PgoDataPackageVersion>
```
Comment (adiaaida) [inline dependencies.props]: We can remove the PgoDataPackageVersion

**Comment by adiaaida** [comment]
Sorry, I just noticed those two things while testing the two changes together!

File: dependencies.props (hunk 1 above)
Comment (hoyosjs) [reply dependencies.props]: Is it a fixed version?

Comment (hoyosjs) [reply dependencies.props]: Or rather, does it build using the old system?

**Comment by adiaaida** [comment]
We should also be able to remove https://github.com/dotnet/coreclr/blob/7920cf173d88bc5c56c7f8ab856dd91627dd0832/build.proj#L40: We upload the packages to dotnetfeed in addition to the myget feed and I'd like to remove the code that uploads it to myget once everyone is onboarded to dotnetfeed for this package (or at least change it to being dotnetfeed)

File: dependencies.props (hunk 1 above)
Comment (adiaaida) [reply dependencies.props]: It is a fixed version, not actively being worked on. It builds manually and a package is uploaded manually.

Comment (hoyosjs) [reply dependencies.props]: Ok, I don't need to revert the update steps then. We should eventually specify this as a toolset dependency in the new format, but I'll leave that for another PR.

Comment (adiaaida) [reply dependencies.props]: Makes sense

**Comment by adiaaida** [comment]
Testing my change with this change went smoothly!

**Comment by hoyosjs** [comment]
To avoid wasting jobs,I'll close this and reopen

**Pull request closed without merging**